### PR TITLE
samples: edac: Add sample documentation

### PR DESCRIPTION
**Pull request by finikorg**
Add EDAC sample documentation.

### CODE REVIEW

**Comment by finikorg** [comment]
I fixed also devmem help string with following PR  #51040.

File: samples/subsys/edac/README.rst [hunk 1]
```diff
@@ -0,0 +1,99 @@
...
+   :goals: run
+   :compact:
+
+Sample output
+*************
+
+Getting help
+============
+
+After application started help can be read with following command::
```
Comment (jhedberg) [inline samples/subsys/edac/README.rst:30]: *the* application

File: samples/subsys/edac/README.rst [hunk 2]
```diff
@@ -0,0 +1,99 @@
+.. _samples_edac:
+
+EDAC Shell Sample
+#################
+
+Overview
+********
+
+This sample demonstrates EDAC driver API in a simple EDAC shell sample.
```
Comment (jhedberg) [inline samples/subsys/edac/README.rst:9]: *the* EDAC driver API

File: samples/subsys/edac/README.rst [hunk 3]
```diff
@@ -0,0 +1,99 @@
...
+
+   uart:~$ edac inject error_type correctable
+   Set injection error type: correctable
+
+Trigger injection with::
+
+   uart:~$ edac inject trigger
+   Triggering injection
+
+Now Read / Write to injection address to trigger Error Injection with
```
Comment (jhedberg) [inline samples/subsys/edac/README.rst:82]: to *the* injection address

File: samples/subsys/edac/README.rst [hunk 4]
```diff
@@ -0,0 +1,99 @@
...
+   Using data width 32
+   Writing value 0xabcd
+
+   uart:~$ devmem 0x1000
+   Mapped 0x1000 to 0x2ffce000
+
+   Using data width 32
+   Read value 0xabcd
+
+We should get following message on screen indicating IBECC event::
```
Comment (jhedberg) [inline samples/subsys/edac/README.rst:97]: *the* following

Comment (jhedberg) [reply samples/subsys/edac/README.rst:97]: indicating *an* IBCC event

**Review by jhedberg — CHANGES_REQUESTED** [review]
Some articles missing in the grammar, otherwise looks good

File: samples/subsys/edac/README.rst (hunk 1 above)
Comment (jhedberg) [reply samples/subsys/edac/README.rst:30]: with *the* following command

File: samples/subsys/edac/README.rst (hunk 4 above)
Comment (finikorg) [reply samples/subsys/edac/README.rst]: Fixed the grammar following this and other comments.

**Review by jhedberg — APPROVED** [review]

**Review by carlescufi — APPROVED** [review]

**Pull request merged**